src/backend/mod.rs

```diff
@@ -1,30 +1,57 @@
-//! This module provides the backend implementations for different terminal libraries.
-//! It defines the [`Backend`] trait which is used to abstract over the specific
-//! terminal library being used.
+//! Provides [`Backend`] implementations for different terminal libraries.
 //!
-//! The following terminal libraries are supported:
-//! - Crossterm (with the `crossterm` feature)
-//! - Termion (with the `termion` feature)
-//! - Termwiz (with the `termwiz` feature)
+//! This module defines the [`Backend`] trait which is used to abstract over the specific terminal
+//! library being used.
+//!
+//! Supported terminal backends:
+//! - [Crossterm]: enable the `crossterm` feature (enabled by default) and use [`CrosstermBackend`]
+//! - [Termion]: enable the `termion` feature and use [`TermionBackend`]
+//! - [Termwiz]: enable the `termwiz` feature and use [`TermwizBackend`]
 //!
 //! Additionally, a [`TestBackend`] is provided for testing purposes.
 //!
 //! # Example
 //!
-//! ```rust
+//! ```rust,no_run
 //! use ratatui::backend::{Backend, CrosstermBackend};
 //!
-//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
-//! let buffer = std::io::stdout();
-//! let mut backend = CrosstermBackend::new(buffer);
+//! let mut backend = CrosstermBackend::on_stdout()?;
 //! backend.clear()?;
-//! # Ok(())
-//! # }
+//! # std::io::Result::Ok(())
 //! ```
 //!
-//! [`Backend`]: trait.Backend.html
-//! [`TestBackend`]: struct.TestBackend.html
-
+//! # Raw Mode
+//!
+//! Raw mode is a mode where the terminal does not perform any processing on the input and output.
+//! This is useful for applications that want to process the input and output themselves. For
+//! example, in raw mode, the terminal will not perform line buffering on the input, so the
+//! application will receive each key press as it is typed instead of waiting for the user to press
+//! enter.
+//!
+//! Each backend handles raw mode differently, so the behavior may vary depending on the backend
+//! being used.
+//!
```

Comment:
Expanding documentation further (aided with ChatGPT)

```suggestion
//! # Raw Mode
//!
//! Raw mode is a mode where the terminal does not perform any processing or handling of the input and output.
//! This means that features such as echoing input characters, line buffering, and special character processing
//! (e.g., CTRL-C for SIGINT) are disabled. This is useful for applications that want to have complete control
//! over the terminal input and output, processing each keystroke themselves.
//!
//! For example, in raw mode, the terminal will not perform line buffering on the input, so the
//! application will receive each key press as it is typed, instead of waiting for the user to press
//! enter. This makes it suitable for real-time applications like text editors, terminal-based games, and more.
//!
//! Each backend handles raw mode differently, so the behavior may vary depending on the backend
//! being used. Be sure to consult the backend's specific documentation for exact details on how it
//! implements raw mode.
```